Automattic/wp-calypso
# Post Editor: Wire up the dialog opt-in button to set the blog's editor to Gutenberg.

DO NOT MERGE: Requires #27892 . It depends on and is branched from #27892 , so that needs to be merged first.

This PR allows the opt-in button to set the site's editor to Gutenberg.

Clicking the "Try the new editor" button will dispatch `EDITOR_TYPE_SET`, which in turn updates the site's editor to Gutenberg over the `/sites/:site/gutenberg` endpoint, and sets `state.sites.items[xxxxxxxxx].editor` to `gutenberg`.

<img width="1196" alt="screen shot 2018-10-16 at 6 39 41 pm" src="https://user-images.githubusercontent.com/349751/47056879-34ad0a80-d173-11e8-9672-6b1c4087f00d.png">

**Testing instructions**
* On this branch, load `/post/:site`.
* Click "Learn more" in the sidebar's opt-in visual.
* Click "Try the new editor".
* Verify a successful API call to `/sites/:site/gutenberg`, and verify the `enable-gutenberg` blog sticker is now applied to the site.
* Verify the update in the store: `state.sites.items[xxxxxxxxx].editor === "gutenberg"`.
